components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -163,71 +128,23 @@ class PlacesApi(path: String) : PlacesManager, AutoCloseable {
     }
 }
 
-internal inline fun <U> rustCall(syncOn: Any, callback: (RustError.ByReference) -> U): U {
-    synchronized(syncOn) {
-        val e = RustError.ByReference()
-        val ret: U = callback(e)
-        if (e.isFailure()) {
-            throw e.intoException()
-        } else {
-            return ret
-        }
-    }
-}
-
-@Suppress("TooGenericExceptionThrown")
-internal inline fun rustCallForString(syncOn: Any, callback: (RustError.ByReference) -> Pointer?): String {
-    val cstring = rustCall(syncOn, callback)
-        ?: throw RuntimeException(
-            "Bug: Don't use this function when you can return" +
-                " null on success."
-        )
-    try {
-        return cstring.getString(0, "utf8")
-    } finally {
-        LibPlacesFFI.INSTANCE.places_destroy_string(cstring)
-    }
-}
-
-internal inline fun rustCallForOptString(syncOn: Any, callback: (RustError.ByReference) -> Pointer?): String? {
-    val cstring = rustCall(syncOn, callback)
-    try {
-        return cstring?.getString(0, "utf8")
-    } finally {
-        cstring?.let { LibPlacesFFI.INSTANCE.places_destroy_string(it) }
-    }
-}
-
 @Suppress("TooGenericExceptionCaught")
-open class PlacesConnection internal constructor(connHandle: Long, uniffiConn: UniffiPlacesConnection) : InterruptibleConnection, AutoCloseable {
-    // As a temp work-around while we uniffi, we actually have 2 references to a PlacesConnection- one
-    // via a handle in the old-school HandleMap, and the other via uniffi.
-    // Each method here will use one or the other, depending on whether it's been uniffi'd or not.
-    protected var handle: AtomicLong = AtomicLong(0)
+open class PlacesConnection internal constructor(uniffiConn: UniffiPlacesConnection) : InterruptibleConnection, AutoCloseable {
     protected var conn: UniffiPlacesConnection
     protected var interruptHandle: SqlInterruptHandle
 
     init {
-        handle.set(connHandle)
-        conn = uniffiConn
         try {
-            interruptHandle = this.conn.newInterruptHandle()
+            interruptHandle = uniffiConn.newInterruptHandle()
         } catch (e: Throwable) {
-            rustCall { error ->
-                LibPlacesFFI.INSTANCE.places_connection_destroy(this.handle.getAndSet(0), error)
-            }
+            uniffiConn.destroy()
```

Comment:
do we need this or will this be destroyed property once it's garbage collected (I am actually not sure 😅)

Reply:
that's a great question :) I believe this just winds up dropping one reference to an `Arc<>`, so it *might* mean things are cleaned up as this is called, and might not. There's no equivalent on iOS, so I'm actually inclined to remove this - if it does turn out to be necessary here, that almost certainly means we need to work out how to do the same thing for iOS. WDYT?

Reply:
Sounds like a plan! either way, it seems like a small detail - I imagine `newInterruptHandle` failing is quite unlikely 😆